src/Lean/Data/HashMap.lean

```diff
@@ -212,6 +212,8 @@ def insertIfNew (m : HashMap α β) (a : α) (b : β) : HashMap α β × Option
 instance : GetElem (HashMap α β) α (Option β) fun _ _ => True where
```

Comment:
Is this the right instance? I had thought that the plan was to make `getElem!` and `getElem?` overridable so that they could do something smarter than "check the side goal, then grab the element", which would allow the element type to efficiently be `β`, so that it would work like array indexing. Did I misunderstand?

Reply:
@david-christiansen The `Option β` instance already pre-existed this PR. Based on this PR's description, I believe the plan is to add the desired `β` instance in a follow-up PR and leave this just as a simple refactor.

Reply:
My bad :) Thanks!